src/lib/utils/functions/getDataFromFromStorage.ts

```diff
@@ -33,3 +38,39 @@ export const getStoredEvaluation = (): Evaluation[] => {
     return []
   }
 }
+
+const getDisplayValueForEntityColumn = (value: string): string => {
+  const entity = getStoredEntityHeaders().find(item => item.id === value)
+  return entity?.name ?? value
+}
+
+const getDisplayValueEvaluationIdColumn = (facetValue: string): string => {
+  const evaluation = getStoredEvaluation().find(item => item.id === facetValue)
+  return evaluation?.name || facetValue
+}
+
+const getDisplayValueUserIdColumn = (facetValue: string): string => {
+  const userProfile = getStoredUserProfiles().find(
+    item => item.ownerId === facetValue,
+  )
+  return userProfile?.userName || facetValue
+}
+
+export const getDisplayValue = (value: string, columnType: ColumnType) => {
+  if (value === SynapseConstants.VALUE_NOT_SET) {
+    return SynapseConstants.FRIENDLY_VALUE_NOT_SET
+  }
+
+  switch (columnType) {
+    case ColumnType.ENTITYID:
+    case ColumnType.ENTITYID_LIST:
+      return getDisplayValueForEntityColumn(value)
+    case ColumnType.USERID:
+    case ColumnType.USERID_LIST:
+      return getDisplayValueUserIdColumn(value)
+    case ColumnType.EVALUATIONID:
+      return getDisplayValueEvaluationIdColumn(value)
+    default:
+      return value
+  }
+}
```

Comment:
Combined the VALUE_NOT_SET rule with fetching by ID from the sessionStorage cache